public/app/features/scenes/dashboard/PanelTimeRange.tsx

```diff
@@ -0,0 +1,140 @@
+  public static Component = PanelTimeRangeRenderer;
+
+  private overrideResult: TimeOverrideResult | null = null;
+
+  public constructor(state: Partial<PanelTimeRangeState> = {}) {
+    const from = state.from ?? 'now-6h';
+    const to = state.to ?? 'now';
+    const timeZone = state.timeZone ?? getTimeZone();
+    const value = evaluateTimeRange(from, to, timeZone);
+    super({ from, to, timeZone, value, ...state });
+  }
+
+  private getTimeOverride(parentTimeRange: TimeRange): TimeOverrideResult {
+    const { timeFrom, timeShift } = this.state;
+    const newTimeData = { timeInfo: '', timeRange: parentTimeRange };
+
+    if (timeFrom) {
+      const timeFromInterpolated = sceneGraph.interpolate(this, this.state.timeFrom);
+      const timeFromInfo = rangeUtil.describeTextRange(timeFromInterpolated);
+
+      if (timeFromInfo.invalid) {
+        newTimeData.timeInfo = 'invalid time override';
+        return newTimeData;
+      }
+
+      // Only evaluate if the timeFrom if parent time is relative
+      if (rangeUtil.isRelativeTimeRange(parentTimeRange.raw)) {
+        newTimeData.timeInfo = timeFromInfo.display;
+        newTimeData.timeRange = {
+          from: dateMath.parse(timeFromInfo.from)!,
+          to: dateMath.parse(timeFromInfo.to)!,
+          raw: { from: timeFromInfo.from, to: timeFromInfo.to },
+        };
+      }
+    }
+
+    if (timeShift) {
+      const timeShiftInterpolated = sceneGraph.interpolate(this, this.state.timeShift);
+      const timeShiftInfo = rangeUtil.describeTextRange(timeShiftInterpolated);
+
+      if (timeShiftInfo.invalid) {
+        newTimeData.timeInfo = 'invalid timeshift';
+        return newTimeData;
+      }
+
+      const timeShift = '-' + timeShiftInterpolated;
+      newTimeData.timeInfo += ' timeshift ' + timeShift;
+      const from = dateMath.parseDateMath(timeShift, newTimeData.timeRange.from, false)!;
+      const to = dateMath.parseDateMath(timeShift, newTimeData.timeRange.to, true)!;
+
+      newTimeData.timeRange = { from, to, raw: { from, to } };
+    }
+
+    return newTimeData;
+  }
+
+  public activate(): void {
+    super.activate();
+
+    const parentTimeRange = this.getParentTimeRange();
+
+    this._subs.add(
+      parentTimeRange.subscribeToState({
+        next: (state) => this.handleParentTimeRangeChanged(state.value),
+      })
+    );
+
+    this.handleParentTimeRangeChanged(parentTimeRange.state.value);
+  }
+
+  private handleParentTimeRangeChanged(parentTimeRange: TimeRange) {
+    this.overrideResult = this.getTimeOverride(parentTimeRange);
+    this.setState({ value: this.overrideResult.timeRange });
+  }
+
+  private getParentTimeRange(): SceneTimeRangeLike {
+    if (!this.parent || !this.parent.parent) {
+      throw new Error('Missing parent');
+    }
+
+    // Need to go up two levels otherwise we will get ourselves
+    return sceneGraph.getTimeRange(this.parent.parent);
```

Comment:
Hit the same problem in Transfromations - but think it's fine. Probably will be not clear for someone seeing this for the first time.